x/twap/client/cli/query.go

```diff
@@ -0,0 +1,138 @@
+package twapcli
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/spf13/cobra"
+
+	gammtypes "github.com/osmosis-labs/osmosis/v12/x/gamm/types"
+	"github.com/osmosis-labs/osmosis/v12/x/twap/client/queryproto"
+	"github.com/osmosis-labs/osmosis/v12/x/twap/types"
+)
+
+// GetQueryCmd returns the cli query commands for this module.
+func GetQueryCmd() *cobra.Command {
+	// Group superfluid queries under a subcommand
+	cmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      fmt.Sprintf("Querying commands for the %s module", types.ModuleName),
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	cmd.AddCommand(GetQueryTwapCommand())
+
+	return cmd
+}
+
+// GetCmdAssetMultiplier returns multiplier of an asset by denom.
+func GetQueryTwapCommand() *cobra.Command {
```

Comment:
nit: don't know if its worth overhead, but would love to have some test cases on this since this includes lot of internal parsing. We could also make it a separate issue for this if u agree with the needs of testing for this

Reply:
Happy if someone else wants to write tests for this, I think of it as pretty low priority.

I'd much rather allocate those mental cycles to getting most CLI code killed / better testing structures. (I believe SDK has progress on better CLI testing structure)